sound/soc/sof/topology.c

```diff
@@ -1354,6 +1354,18 @@ int sof_load_pipeline_ipc(struct snd_sof_dev *sdev,
 
 	mutex_lock(&sdev->cores_mask_mutex);
 
+	/* already powered on, return */
+	if (atomic_inc_return(&sdev->core_refs[pipeline->core]) != 1) {
```

Comment:
same here, you are incrementing the values under the cores_mask_mutex protection, why is the atomic_inc needed?